src/xtgeo/surface/regular_surface.py

```diff
@@ -100,11 +108,23 @@ def surface_from_file(mfile, fformat=None, template=None, values=True, engine="c
     """
 
     return RegularSurface._read_file(
-        mfile, fformat=fformat, load_values=values, engine=engine, template=template
+        mfile,
+        fformat=fformat,
+        load_values=values,
+        engine=engine,
+        template=template,
+        dtype=dtype,
     )
 
 
-def surface_from_roxar(project, name, category, stype="horizons", realisation=0):
+def surface_from_roxar(
+    project,
+    name,
+    category,
+    stype="horizons",
+    realisation=0,
+    dtype: Optional[Union[np.float64, np.float32]] = np.float64,
```

Comment:
The typing yu want here is `Union[Type[np.float64], Type[np.float32]]`. `Optional` means the value can be `None` having type np.float64 means the value is e.g. `0.1`.

Reply:
Of course; thanx :-)